### PR TITLE
Add errorhandler

### PR DESCRIPTION
**Pull request by msm-faraj**
This pr aims to:
add accountController error paths to errorHandler middleware
add authController error paths to errorHandler middleware

### CODE REVIEW

File: src/controllers/account/accountController.js [hunk 1]
```diff
@@ -12,11 +13,8 @@ class AccountController {
     const user = await this.User.findOne({
       where: { id: req.authenticatedUser.id },
     });
-    if (!user) {
-      return res
-        .status(httpStatus.UNAUTHORIZED)
-        .send("You are not authenticated");
-    }
+    if (!user) throw AuthenticationError("You are not authenticated");
```
Comment (pouyapourbaba) [inline src/controllers/account/accountController.js:16]:
```suggestion
    if (!user) throw new AuthenticationError("You are not authenticated");
```

File: src/controllers/account/accountController.js [hunk 2]
```diff
@@ -51,9 +48,8 @@ class AccountController {
       where: { id: req.params.id },
     });
 
-    if (!account) {
-      return res.status(httpStatus.NOT_FOUND).send("Account not found");
-    }
+    if (!account) throw NotFoundError("Account not found");
```
Comment (pouyapourbaba) [inline src/controllers/account/accountController.js:51]:
```suggestion
    if (!account) throw new NotFoundError("Account not found");
```

File: src/lib/errors.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+class ConflictError extends Error {
+  constructor(message) {
+    super(message);
+    this.status = 409;
+    this.name = "ConflictError";
+  }
+}
+
+class BadRequest extends Error {
```
Comment (pouyapourbaba) [inline src/lib/errors.js:33]:
```suggestion
class ValidationError extends Error {
```

File: src/lib/errors.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  constructor(message) {
+    super(message);
+    this.status = 400;
+    this.name = "BadRequest";
```
Comment (pouyapourbaba) [inline src/lib/errors.js:37]:
```suggestion
    this.name = "ValidationError";
```

Comment (msm-faraj) [reply src/lib/errors.js:37]: done

File: src/lib/errors.js (hunk 3 above)
Comment (msm-faraj) [reply src/lib/errors.js:33]: done

File: src/controllers/account/accountController.js (hunk 2 above)
Comment (msm-faraj) [reply src/controllers/account/accountController.js:51]: done

File: src/controllers/account/accountController.js (hunk 1 above)
Comment (msm-faraj) [reply src/controllers/account/accountController.js:16]: done

**Review by pouyapourbaba — APPROVED** [review]

**Pull request merged**